contracts/minter/src/msg.rs

```diff
@@ -0,0 +1,68 @@
+use cosmwasm_std::{Addr, Coin};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cw_utils::Expiration;
+use sg721::msg::InstantiateMsg as Sg721InstantiateMsg;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {
+    pub base_token_uri: String,
+    pub num_tokens: u64,
+    pub sg721_code_id: u64,
+    pub sg721_instantiate_msg: Sg721InstantiateMsg,
+    pub start_time: Option<Expiration>,
+    pub per_address_limit: Option<u32>,
+    pub unit_price: Coin,
+    pub whitelist: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    Mint {},
+    SetWhitelist { whitelist: String },
+    UpdateStartTime(Expiration),
+    UpdatePerAddressLimit { per_address_limit: u32 },
+    MintTo { recipient: String },
```

Comment:
I think you should add a /// comment to differentiate `MintTo` and `MintFor`. Easy to understand without looking at code.